tasks/lib/DocumentParser.js

```diff
@@ -81,8 +90,14 @@ class DocumentParser {
         }
       }
       if (line.trim().endsWith('-->')) {
-        this.inComment = false;
-        this.currentTag = this.nextTag(i);
+        if (line.trim().endsWith('--->')) {
+          this.inMetadata = false;
+          this.document.metadata = JSON.parse(this.metadata);
+          this.currentTag = this.nextTag(i);
+        } else {
+          this.inComment = false;
+          this.currentTag = this.nextTag(i);
```

Comment:
move `this.currentTag = this.nextTag(i);` in the outer diff so that we don't need to repeat it